ReactNativeClient/lib/services/rest/Api.js

```diff
@@ -451,6 +451,23 @@ class Api {
 			return note;
 		}
 
+		if (request.method === 'PUT') {
+			const note = await Note.load(id);
+			const requestBody = JSON.parse(request.body);
+
+			if (!note) throw new ErrorNotFound();
+
+			let newNote = Object.assign({}, note, request.bodyJson(this.readonlyProperties('PUT')));
+			newNote = await Note.save(newNote, { userSideValidation: true });
```

Comment:
What is the effect of user-side validation here?

Reply:
The code I have written here was actually refactored from the `defaultAction_` function which was previously called in the case of `PUT` methods. 

To be honest, I do not know exactly know what user-side validation does here. But since it was already present in the code that was executed (before this fix), I decided to leave it untampered in order not to break any functionality.